### ISSUE TITLE
[COMPILER] Seg fault related to locals in ensure block

### ISSUE THREAD

**Issue by jemmaissroff** [ruby/prism#1994]
Minimal repro:

```
def test
ensure
  {}.each do |key, value|
    {}[key] = value
  end
end
```

Failing test: `RUBY_ISEQ_DUMP_DEBUG=prism make test/irb/test_locale.rb`

**Comment by jemmaissroff** [comment]
The same seems to happen for a rescue block:

```
begin
rescue
  args.each do |key, value|
    tmp[key] = 1
  end
end
```

**Comment by jemmaissroff** [comment]
Another example:

```
begin
ensure
  a do
    b = 1
    c = 2
  end
end
```

`RUBY_ISEQ_DUMP_DEBUG=prism make test/openssl/test_asn1.rb`

**Comment by eightbitraptor** [comment]
The first 3 should be fixed by https://github.com/ruby/ruby/pull/9144

The last one looks like a different problem though. I've created https://github.com/ruby/prism/issues/1999

**Comment by eightbitraptor** [comment]
I'm closing this as fixed since #9144 was merged containing the first three test cases (now passing), and the final one has been represented in #1999
